src/ai/backend/manager/models/group.py

```diff
@@ -116,31 +116,59 @@
 )
 
 
+def _build_group_query(cond, domain_name):
+    query = (
+        sa.select([groups.c.id])
+        .select_from(groups)
+        .where(
+            cond & (groups.c.domain_name == domain_name),
+        )
+    )
+    return query
+
+
 async def resolve_group_name_or_id(
     db_conn: SAConnection,
     domain_name: str,
     value: Union[str, uuid.UUID],
 ) -> Optional[uuid.UUID]:
-    if isinstance(value, str):
-        query = (
-            sa.select([groups.c.id])
-            .select_from(groups)
-            .where(
-                (groups.c.name == value) & (groups.c.domain_name == domain_name),
-            )
-        )
-        return await db_conn.scalar(query)
-    elif isinstance(value, uuid.UUID):
-        query = (
-            sa.select([groups.c.id])
-            .select_from(groups)
-            .where(
-                (groups.c.id == value) & (groups.c.domain_name == domain_name),
-            )
-        )
-        return await db_conn.scalar(query)
-    else:
-        raise TypeError("unexpected type for group_name_or_id")
+    match value:
+        case uuid.UUID():
+            cond = groups.c.id == value
+        case str():
+            cond = groups.c.name == value
+        case _:
+            raise TypeError("unexpected type for group_name_or_id")
+    query = _build_group_query(cond, domain_name)
+    return await db_conn.scalar(query)
+
+
+async def resolve_groups(
+    db_conn: SAConnection,
+    domain_name: str,
+    values: Iterable[str | uuid.UUID],
```

Comment:
How about using `typing.overload`?
Also, let's do `Iterable[str] | Iterable[uuid.UUID]` instead of `Iterable[str | uuid.UUID]`.